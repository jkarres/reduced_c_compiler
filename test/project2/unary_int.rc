// testing unary-sign + ints

const int ci = -5;
int i = +5;

function : void main() {
  const int cj = +10;
  const int j = -10;

  cout << 1 << " " << ci << " " << +ci << " " << -ci << endl;
  cout << 2 << " " << i << " " << +i << " " << -i << endl;
  cout << 3 << " " << cj << " " << +cj << " " << -cj << endl;
  cout << 4 << " " << j << " " << +j << " " << -j << endl;
  cout << 5 << " " << 7 << " " << +7 << " " << -7 << endl;
  cout << 6 << " " << (-2 + - -3) << " " << +(-2 + - -3) << " " << -(-2 + - -3) << endl;
}