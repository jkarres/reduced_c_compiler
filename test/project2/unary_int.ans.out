1 -5 -5 5
2 5 5 -5
3 10 10 -10
4 -10 -10 10
5 7 7 -7
6 1 1 -1
